src/server/auth.ts

```diff
@@ -48,12 +47,19 @@ export const writeAccounts = async (accounts: Account[]): Promise<void> => {
 
 /** simple middleware to check whether the user is logged in */
 export const requireLogin = (req, res, next) => {
-    if (!req.session || !req.session.user) {
-        log.debug('not logged in');
-        res.redirect('/login');
-    } else {
-        next();
-    }
+  if (req.session && req.session.user) {
+    next();
+    return;
+  }
+  log.debug('not logged in', req.url);
+  if (req.xhr || req.headers.accept.indexOf('json') > -1) {
```

Comment:
This seems unnecessarily complicated. We control the other side, so guessing whether the client accepts json is not required, we can just enforce it, no? My sense is that we can just always use `res.direct` now that we have a proper login route.